File: leetcode/all/typescript/2006.差的绝对值为-k-的数对数目.ts
/*
 * @lc app=leetcode.cn id=2006 lang=typescript
 *
 * [2006] 差的绝对值为 K 的数对数目
 */

// @lc code=start
function countKDifference(nums: number[], k: number): number {
    let ans = 0
    const map = {}
    for (const num of nums) {
        ans += (map[num + k] || 0) + (map[num - k] || 0)
        map[num] = (map[num] || 0) + 1
    }
    return ans
}
// @lc code=end

File: leetcode/all/python3/2000-2099/2006.差的绝对值为-k-的数对数目.py
#
# @lc app=leetcode.cn id=2006 lang=python3
#
# [2006] 差的绝对值为 K 的数对数目
#

# @lc code=start
from collections import Counter
from typing import List


class Solution:
    def countKDifference(self, nums: List[int], k: int) -> int:
        ans = 0
        counter = Counter()
        for num in nums:
            ans += counter[num + k] + counter[num - k]
            counter[num] += 1
        return ans


# @lc code=end
